# 5. AspC simulation of heterologous expression

In [1]:
%run 1_Objective_function.ipynb

EX_asp__L_e EX_asp__L_e: asp__L_e -->  L-Aspartate exchange
ASPtex ASPtex: asp__L_e <=> asp__L_p L-aspartate transport via diffusion (extracellular to periplasm)
0.87699721425716
0.87392
4.596027557488181


_Pseudomonas aeruginosa_'s AspC has a 30% higher kcat than E. coli's AspC for L-aspartate production. Therefore, we will change the lower and upper bounds of the kcat of AspC to 30% higher than the default.

In [10]:
original_solution = model.optimize()
print("Original Objective Value for Biomass:", original_solution.objective_value)

Original Objective Value for Biomass: 0.87392


In [25]:
reaction_id = "ASPTA"
# Check if the reaction is in the model
if reaction_id in [rxn.id for rxn in model.reactions]:
    # Get the reaction object
    reaction = model.reactions.get_by_id(reaction_id)
    print(f"Reaction {reaction_id} is present in the model.")
    print("Reaction details:", reaction)
    print(f"Lower bound: {reaction.lower_bound}")
    print(f"Upper bound: {reaction.upper_bound}")

Reaction ASPTA is present in the model.
Reaction details: ASPTA: akg_c + asp__L_c <=> glu__L_c + oaa_c
Lower bound: -1000.0
Upper bound: 1000.0


In [26]:
# Check if the reaction is in the model
if reaction_id in [rxn.id for rxn in model.reactions]:
    # Run Flux Balance Analysis
    solution = model.optimize()

    # Get the flux of the reaction before changing bounds
    initial_flux = solution.fluxes[reaction_id]
    print(f"Initial flux for {reaction_id}: {initial_flux}")

    # Get the reaction object
    reaction = model.reactions.get_by_id(reaction_id)

    # Current bounds
    current_lower_bound = reaction.lower_bound
    current_upper_bound = reaction.upper_bound

    # Increase bounds by 30%
    new_lower_bound = current_lower_bound * 1.30
    new_upper_bound = current_upper_bound * 1.30

    # Update the bounds in the model
    reaction.lower_bound = new_lower_bound
    reaction.upper_bound = new_upper_bound

    # Run FBA again and print new flux
    new_solution = model.optimize()
    new_flux = new_solution.fluxes[reaction_id]
    print(f"New flux for {reaction_id}: {new_flux}")

Initial flux for ASPTA: -2.6228433417913637
New flux for ASPTA: -2.622843341791384


The change in flux for the ASPTA reaction before and after adjusting the bounds, (-2.6228433417913637 initially and -2.622843341791384 after the change), is extremely small, so we will not change the bounds of the ASPTA reaction.